modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/exportimport/portlet/preferences/processor/DLExportImportPortletPreferencesProcessor.java

```diff
@@ -340,6 +340,25 @@ public PortletPreferences processImportPortletPreferences(
 				}
 			}
 		}
+		else {
+			try {
+				long selectedRepositoryId = GetterUtil.getLong(
+					portletPreferences.getValue("selectedRepositoryId", null));
+
+				if (selectedRepositoryId ==
```

Comment:
This works smoothly when we have a folder of the current site previously selected on the D&M and then stage the site.
But if we select a folder of the site on the configuration of the D&M widget **when staged**, the selectedRepositoryId it's the staged one so the https://issues.liferay.com/browse/LPS-143128 and https://issues.liferay.com/browse/LPS-143130 still are happening (on this case) because it is not executing this code

A solution is to disable staging and enabling staging, but maybe it's not an option

We can pass this PR and open a new bug, it's up to you.

Reply:
I'm working on that under https://issues.liferay.com/browse/LPS-145279; I need this merged first, because I need to include relatively big changes in LPS-145279 (e.g. added change set support and two upgrades), and I prefer to send it as two smaller changes.